### PR TITLE
fix: use .watch() to disable update btn

### PR DESCRIPTION
**Pull request by Udit-takkar**
refs: #5424

https://user-images.githubusercontent.com/53316345/201348407-6136c4ab-4702-4389-95f9-86219027c257.mov

this was suggested by @Jaibles here https://threads.com/34433127474?s=ucqt4HpZQcM3tsQmy5m8G5

### CODE REVIEW

File: apps/web/pages/settings/my-account/profile.tsx [hunk 1]
```diff
@@ -156,7 +156,7 @@ const ProfileView = () => {
   const {
     reset,
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (zomars) [inline apps/web/pages/settings/my-account/profile.tsx:158]:
```suggestion
```

**Review by zomars — CHANGES_REQUESTED** [review]
No need for use `watch` we can just use the `isDirty` property from `formState`

**Review by zomars — COMMENTED** [review]
LGTM

**Review by zomars — APPROVED** [review]

**Comment by Udit-takkar** [comment]
@zomars  we don't want the update button to be enabled when username is changed because there is already a separate update button for it. isDirty is checking all the fields (including username)

  is there any other way other than .watch() to achieve this?

Screenshot 2022-11-12 at 3 47 12 PM

**Comment by zomars** [comment]
We should encapsulate the username field in another form. So it can have it's own formstate

**Pull request merged**